js/foam/core/Enum.js

```diff
@@ -183,7 +206,12 @@ public enum <%= this.name %> {
     var value = this.values[i];
     if ( value.javaSource ) { value.javaSource(out); }
     else {
-%>  <%= value.name %>(<%= value.index %>, <%= value.value %>)<%
+      if (this.androidResource) {
+        var label = this.androidResource + "." + this.name + "_" + value.name
+      } else {
+        var label = value.label ? value.label : value.name
```

Comment:
What about adding a "label" property to EnumValue, with defaultValueFn: function() { return this.name; }

then all these conditionals of value.label ? value.label : value.name, can just be value.label